### PR TITLE
Fixes #9077: Inventory "toInt" error when parsing report with VIRTUALMEMORY = 2297795780

### PR DESCRIPTION
**Pull request by fanf**
https://www.rudder-project.org/redmine/issues/9077

### CODE REVIEW

File: inventory-fusion/src/main/scala/com/normation/inventory/provisioning/fusion/FusionReportUnmarshaller.scala [hunk 1]
```diff
@@ -76,11 +76,27 @@ class FusionReportUnmarshaller(
...
     case s => Some(s.trim().replaceAll("""[\p{Blank}]+"""," "))
   }
 
+  //extremelly specialized convert used for optionnal field only, that
+  //log the error in place of using a box
+  private[this] def convert[T](input: Option[String], conv: String => T): Option[T] = {
+    try {
+      input.map( s => conv(s))
+    } catch {
+      case ex: Exception => Failure(ex.getMessage, Full(ex), Empty)
```
Comment (VinceMacBuche) [inline inventory-fusion/src/main/scala/com/normation/inventory/provisioning/fusion/FusionReportUnmarshaller.scala]: You said in comment, that you would log the error in place, put i see no log here...

And as it use an option, The failure will be converted to None so you lost the failure (maybe returning a None would be better then

**Comment by fanf** [comment]
PR rebased

File: inventory-fusion/src/main/scala/com/normation/inventory/provisioning/fusion/FusionReportUnmarshaller.scala (hunk 1 above)
Comment (fanf) [inline inventory-fusion/src/main/scala/com/normation/inventory/provisioning/fusion/FusionReportUnmarshaller.scala]: Yeah, the intent was here, my fingers seem to have hold a strike at some point.

**Comment by FlorianHeigl** [comment]
+1 :)

File: inventory-fusion/src/main/scala/com/normation/inventory/provisioning/fusion/FusionReportUnmarshaller.scala [hunk 2]
```diff
@@ -76,11 +76,29 @@ class FusionReportUnmarshaller(
...
   }
 
+  //extremelly specialized convert used for optionnal field only, that
+  //log the error in place of using a box
+  private[this] def convert[T](input: Option[String], tag: String, format: String, conv: String => T): Option[T] = {
+    try {
+      input.map( s => conv(s))
+    } catch {
+      case ex: Exception =>
+        logger.warn(s"Ignoring '${tag}' content because it can't be convert to ${}. Error is: ${ex.getMessage}")
```
Comment (VinceMacBuche) [inline inventory-fusion/src/main/scala/com/normation/inventory/provisioning/fusion/FusionReportUnmarshaller.scala]: It miss something in the second $ ? param 'format' ?

Comment (VinceMacBuche) [inline inventory-fusion/src/main/scala/com/normation/inventory/provisioning/fusion/FusionReportUnmarshaller.scala]: converted

**Comment by fanf** [comment]
PR rebased

**Pull request merged**